src/game/crossteaser/mod.rs

```diff
@@ -103,45 +658,178 @@ impl Game for Session {
     }
 
     fn info(&self) -> GameData {
-        todo!()
+        let var = match &self.variant {
+            None => VARIANT_DEFAULT,
+            Some(v) => v,
+        };
+        GameData {
+            variant: var.to_owned(),
+
+            name: NAME,
+            authors: AUTHORS,
+            about: ABOUT,
+
+            variant_protocol: VARIANT_PROTOCOL,
+            variant_pattern: VARIANT_PATTERN,
+            variant_default: VARIANT_DEFAULT,
+
+            state_default: STATE_DEFAULT,
+            state_pattern: STATE_PATTERN,
+            state_protocol: STATE_PROTOCOL,
+        }
     }
 
     fn solve(&self, mode: IOMode, method: SolutionMode) -> Result<()> {
         todo!()
+        /*
+        solver::algorithm::strong::puzzle::dynamic_solver::<Self>(self, mode)
+            .context("Failed solver run.")
+            */
     }
 }
 
 /* TRAVERSAL IMPLEMENTATIONS */
 
 impl DTransition for Session {
     fn prograde(&self, state: State) -> Vec<State> {
-        todo!()
+        let s: UnhashedState = self.unhash(state);
```

Comment:
In the spirit of the other comment I left about making all of this a single function that takes some `enum Move`, consider the following abstraction that saves us a lot of this logic:

```
fn prograde(&self, state: State) -> Vec<State> {
  
  // Could even be a const item MOVES
  let moves: [Move; 6] = [
    Move::UP,
    ...
  ];
  
  let unhashed = self.unhash(state);
  moves.iter()
    .map(|m| self.transition(unhashed))
    .filter(|s| s.is_some())
    .map(|s| s.unwrap())
    .map(|s| self.hash(s))
    .collect::<Vec<State>>()
}

fn transition(&self, state: UnhashedState, move: Move) -> Option<State> {
  ... // This handles whether or not the move can actually be made.
}
```

You can think of how this would be easier to scale to many-free-slot variants.